src/compas_timber/_fabrication/step_joint_notch.py

```diff
@@ -0,0 +1,796 @@
+        super(StepJointNotch, self).__init__(**kwargs)
+        self._orientation = None
+        self._start_x = None
+        self._start_y = None
+        self._strut_inclination = None
+        self._notch_limited = None
+        self._notch_width = None
+        self._step_depth = None
+        self._heel_depth = None
+        self._strut_height = None
+        self._step_shape = None
+        self._mortise = None
+        self._mortise_width = None
+        self._mortise_height = None
+
+        self.orientation = orientation
+        self.start_x = start_x
+        self.start_y = start_y
+        self.strut_inclination = strut_inclination
+        self.notch_limited = notch_limited
+        self.notch_width = notch_width
+        self.step_depth = step_depth
+        self.heel_depth = heel_depth
+        self.strut_height = strut_height
+        self.step_shape = step_shape
+        self.mortise = mortise
+        self.mortise_width = mortise_width
+        self.mortise_height = mortise_height
+
+    ########################################################################
+    # Properties
+    ########################################################################
+
+    @property
+    def params_dict(self):
+        return StepJointNotchParams(self).as_dict()
+
+    @property
+    def orientation(self):
+        return self._orientation
+
+    @orientation.setter
+    def orientation(self, orientation):
+        if orientation not in [OrientationType.START, OrientationType.END]:
+            raise ValueError("Orientation must be either OrientationType.START or OrientationType.END.")
+        self._orientation = orientation
+
+    @property
+    def start_x(self):
+        return self._start_x
+
+    @start_x.setter
+    def start_x(self, start_x):
+        if start_x > 100000.0 or start_x < -100000.0:
+            raise ValueError("StartX must be between -100000.0 and 100000.")
+        self._start_x = start_x
+
+    @property
+    def start_y(self):
+        return self._start_y
+
+    @start_y.setter
+    def start_y(self, start_y):
+        if start_y > 50000.0 or start_y < -50000.0:
+            raise ValueError("StartY must be between -50000.0 and 50000.")
+        self._start_y = start_y
+
+    @property
+    def strut_inclination(self):
+        return self._strut_inclination
+
+    @strut_inclination.setter
+    def strut_inclination(self, strut_inclination):
+        if strut_inclination < 0.1 or strut_inclination > 179.9:
+            raise ValueError("StrutInclination must be between 0.1 and 179.9.")
+        self._strut_inclination = strut_inclination
+
+    @property
+    def notch_limited(self):
+        return self._notch_limited
+
+    @notch_limited.setter
+    def notch_limited(self, notch_limited):
+        if notch_limited not in ["no", "yes"]:
```

Comment:
Yeah I agree with your point, it's the way to keep it clean. Since we are also using Enums we might as well maintain them separate and translate everything in the parameters class

Reply:
Do you think it's best we implement:

1. a generic boolean Enum that can be used in these yes/no cases (ie. `BTLxBoolean.YES`,` BTLxBoolean.NO`)
2. a specific Enum for each case (ie. `NotchLimited.YES`, `NotchLimited.NO`) 
3. just use a boolean and translate it then in the parameter class?